packages/components/src/presentation-components/properties/UniquePropertyValuesTargetSelector.tsx

```diff
@@ -0,0 +1,154 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { useCallback, useEffect, useState } from "react";
+import { ActionMeta, MultiValue } from "react-select";
+import { PropertyDescription, PropertyValue, PropertyValueFormat } from "@itwin/appui-abstract";
+import { PropertyFilterBuilderRuleValueProps } from "@itwin/components-react";
+import { IModelConnection } from "@itwin/core-frontend";
+import { ContentSpecificationTypes, Descriptor, DisplayValueGroup, FieldDescriptor, KeySet, Ruleset, RuleTypes, Value } from "@itwin/presentation-common";
+import { Presentation } from "@itwin/presentation-frontend";
+import { AsyncMultiTagSelect } from "../instance-filter-builder/MultiTagSelect";
+import { PRESENTATION_TARGETS_BATCH_SIZE } from "./UseNavigationPropertyTargetsLoader";
+import { translate } from "../common/Utils";
+
+/** @internal */
+export function UniquePropertyValuesTargetSelector(props: PropertyFilterBuilderRuleValueProps & { imodel: IModelConnection; descriptor: Descriptor }) {
+  const { imodel, descriptor, property, onChange, value } = props;
+  const [selectedTarget, setSelectedTarget] = useState<DisplayValueGroup[] | undefined>(() => getUniqueValueTargetFromProperty(value));
+  const [fieldDescriptor, setFieldDescriptor] = useState<FieldDescriptor | undefined>(() =>
+    descriptor.fields.find((field) => field.label === property.displayLabel)?.getFieldDescriptor(),
+  );
+
+  useEffect(() => {
+    setFieldDescriptor(descriptor.fields.find((field) => field.label === property.displayLabel)?.getFieldDescriptor());
+  }, [descriptor, property]);
+
+  useEffect(() => {
+    setSelectedTarget(getUniqueValueTargetFromProperty(value));
+  }, [value]);
+
+  const onValueChange = useCallback(
+    (newValue: MultiValue<DisplayValueGroup>, _: ActionMeta<DisplayValueGroup>) => {
+      setSelectedTarget(newValue.map((item) => item));
+      if (newValue.length === 0) {
+        onChange({
+          valueFormat: PropertyValueFormat.Primitive,
+          displayValue: undefined,
+          value: undefined,
+        });
+      } else {
+        onChange({
+          valueFormat: PropertyValueFormat.Primitive,
+          displayValue: JSON.stringify(newValue.map((item) => item.displayValue)),
+          value: stringifyGroupRawValuesTo(newValue),
+        });
+      }
+    },
+    [onChange],
+  );
+
+  const ruleset = useUniquePropertyValuesRuleset({ property, fieldDescriptor });
+  const loadTargets = useUniquePropertyValuesLoader({ imodel, ruleset, fieldDescriptor });
+
+  return (
+    <AsyncMultiTagSelect
+      value={selectedTarget ?? null}
+      loadOptions={async (inputValue, options) => loadTargets(inputValue, options.length)}
```

Comment:
You can remove first argument from `loadTargets` as it is not used.

Reply:
fixed